compiler/src/dotty/tools/dotc/transform/LetOverApply.scala

```diff
@@ -0,0 +1,37 @@
+package dotty.tools
+package dotc
+package transform
+
+import core._
+import MegaPhase._
+import Contexts.Context
+import Symbols._, Decorators._, Types._
+import ast.Trees._
+import dotty.tools.dotc.ast.tpd
+
+
+import scala.collection.immutable.::
+
+
+/** Rewrite `{ stats; expr}.f(args) }` to `{ stats; expr.f(args) }` before
+ *  proceeding, but leave closures alone. This is necessary to be able to
+ *  collapse applies of IFTs.
```

Comment:
Since the collapsing of IFTs is done in Erasure, this phase should be added to `Erasure#runsAfter`

Reply:
I guess erasure needs to run after about all the phases that precede it, yet it lists only two. Maybe we should drop the pretense and just take erasure as a special watershed that divides phases into two.

Reply:
Some phases might be able to run either before or after erasure, but it's true that we don't try to list them all. We can at least document that this phase is here for the benefit of erasure in particular, otherwise it can be hard to find where the mentioned collapse of IFT is:
```suggestion
 *  collapse applies of IFTs (this is done in Erasure).
```